libr/asm/arch/jdh8/jdh8dis.c

```diff
@@ -0,0 +1,140 @@
+/* radare - LGPL3 - Copyright 2021 - condret, slowhand99 */
+
+#include <r_types.h>
+#include <r_util.h>
+#include <r_asm.h>
+
+static const int jdh_len[16] = {
+	2, 3, 3, 1, 1, 3, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2
+};
+
+static const char reg[6] = { 'A', 'B', 'C', 'D', 'L', 'H' };
+
+static int jdh_get_ins_len(ut8 hex) {
+	ut8 high = (hex & 0xf0) >> 4;
+	int ret = jdh_len[high];
+	if (ret == 3) {
+		ret = (hex & 8) ? 2 : 3;
+	}
+	if (ret == 1) {
+		ret = (hex & 8) ? 1 : 2;
+	}
+	return ret;
+}
+
+static int jdh8Disass(RAsmOp *op, const ut8 *buf, int len) {
+	int rlen = jdh_get_ins_len (*buf);
+	ut8 high = (*buf & 0xf0) >> 4;
+	ut8 low = (*buf & 0xf);
+	const char *buf_asm = "invalid";
+	if (rlen > len) {
+		return op->size = 0;
+	}
+	switch (high) {
+	case 0:
+		buf_asm = (low & 8)? sdb_fmt ("mw %c, %c", reg[low & 7], reg[buf[1]]):
```

Comment:
sdb_fmt is not thread safe and we are trying to get rid of it across all the r2codebase, there are alternatives like snprintf or r_str_newf in case it must be held of the heap. in this case it would work better to have a local buffer of 64 bytes in this function and use snprintf like this:

```
char buf[64];
buf_asm = snprintf (buf, sizeof (buf), "mw %c, %c", ...);
..
```

Reply:
you can add a #define to avoid having to pass buf and sizeof(buf) all the time so its easier for the eye 😄

Reply:
> #define to avoid having to pass buf and sizeof(buf)

How? Can you give an example?

Reply:
but dont u think is better to rewrite the plugin as an analysis one so it doesnt needs to duplicate the logic for the analysis? also, are there plans to implement the assembler fr this arch too?